### PR TITLE
[DOCS] Reformat match query

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the `match` query to use the new query format.

This creates separate sections for the example request, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

### Preview
http://elasticsearch_45152.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-match-query.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine  run  elasticsearch-ci/docs-check elasticsearch-ci/oss-distro-docs

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/oss-distro-docs

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/query-dsl/match-query.asciidoc [hunk 1]
```diff
@@ -4,28 +4,144 @@
...
+See <<query-dsl-match-query-zero>> for an example.
+--
+
+
+[[match-query-notes]]
+==== Notes
 
 [[query-dsl-match-query-boolean]]
-==== match
+===== How the match query works
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/match-query.asciidoc]: Line 154 (note the slight . change in structure, `message` is the field name):	change in structure, `message` is the field name)
 Should we remove this part? As this is the structure we are showing from the beginning

File: docs/reference/query-dsl/match-query.asciidoc [hunk 2]
```diff
@@ -4,28 +4,144 @@
 <titleabbrev>Match</titleabbrev>
 ++++
 
+Returns documents that approximately match a provided text, number, or date.
 
-`match` queries accept text/numerics/dates, analyzes
-them, and constructs a query. For example:
+The `match` query is the standard query for performing a full-text search,
+including options for fuzzy and proximity matching.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/match-query.asciidoc]: > proximity matching
 
I don't think match query supports any proximity matching.

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig.  match query is tricky, I left a couple of comments! good work!

File: docs/reference/query-dsl/match-query.asciidoc [hunk 3]
```diff
@@ -4,28 +4,144 @@
 <titleabbrev>Match</titleabbrev>
 ++++
 
+Returns documents that approximately match a provided text, number, or date.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/match-query.asciidoc]: Changed to the suggested text with 438cbee.

File: docs/reference/query-dsl/match-query.asciidoc [hunk 4]
```diff
@@ -4,28 +4,144 @@
...
+==== Example request
 
 [source,js]
 --------------------------------------------------
 GET /_search
 {
     "query": {
         "match" : {
-            "message" : "this is a test"
+            "message" : {
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/match-query.asciidoc]: I added this in its own section with ba7bf5b.

File: docs/reference/query-dsl/match-query.asciidoc [hunk 5]
```diff
@@ -4,28 +4,144 @@
...
+`<field>`::
+(Required, object) Field you wish to search.
+
+
+[[match-field-params]]
+==== Parameters for `<field>`
+`query`::
++
+--
+(Required, string) Text, number, or date you wish to find in the provided
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/match-query.asciidoc]: Fixed with db8c1aa.

File: docs/reference/query-dsl/match-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/match-query.asciidoc]: Good catch! Fixed with 10ce6e345e9.

File: docs/reference/query-dsl/match-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/match-query.asciidoc]: Removed with 461b17cafebca55778751774713cf4562bb1e089.

**Comment by jrodewig** [comment]
Thanks again for your review @mayya-sharipova!

**Comment by colings86** [comment]
It looks like the backporting is complete here so I have removed the backport pending label while generating 7.4.0 release notes. Please let me know if this is wrong

**Comment by jrodewig** [comment]
That's correct. Thanks @colings86.

**Pull request merged**